### PR TITLE
Line diffs

### LINKED ISSUE #173: add highlighting to single-line diff lines
Examples from Desktop Classic on Windows:

[Classic Logic](https://github.com/github/Windows/blob/master/GitHub/Models/TextFileDiff.cs#L89-L119)

### PR DESCRIPTION
**Pull request by joshaber**
Fixes #173

This is a pretty direct port of Mac Classic's previous algorithm: https://github.com/github/Mac/blob/e5d08b48aacfa9737f3953cf42cc105f399adc7a/Classes/GHTextDiffLineViewModel.m#L170-L193

It looks like Windows' approach was fairly different. I'm not sure which is _better_, but the Mac one was done by _**the**_ `@robrix`.

screen shot 2017-08-16 at 5 15 19 pm

### CODE REVIEW

File: app/src/ui/diff/diff-mode.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ const TokenNames: { [key: string]: string | null } = {
   '@': 'diff-hunk',
 }
 
-/** Get the mode for diffs. */
+/** Get the mod for diffsssss. */
```
Comment (shiftkey) [inline app/src/ui/diff/diff-mode.ts]: lol

Comment (joshaber) [inline app/src/ui/diff/diff-mode.ts]: :shipit:

File: app/styles/_variables.scss [hunk 2]
```diff
@@ -237,11 +237,13 @@
   --diff-add-background-color: $green-000;
   --diff-add-border-color: $green-200;
   --diff-add-gutter-color: $green-200;
+  --diff-add-inner-background-color: #acf2bd;
```
Comment (joshaber) [inline app/styles/_variables.scss]: I took these colors straight from dotcom.

**Comment by joshaber** [comment]
This is ready 🙏 🇵🇾

**Comment by shiftkey** [comment]
> by **the** `@robrix`.

**Comment by shiftkey** [comment]
There's an interesting quirk here where the line diff should be split into multiple sections in the app but is drawn as one - here's [this commit](https://github.com/desktop/dugite-native/commit/5810f66b1c753253568e8a819a4d23c50305b5eb) illustrated in both cases:

**Comment by joshaber** [comment]
>There's an interesting quirk here where the line diff should be split into multiple sections in the app but is drawn as one

Yeah, that's a known limitation... the way the heuristic works is by finding the longest common substring starting from the beginning and end. And then anything in between the beginning and end ranges is considered the change.

I opened https://github.com/desktop/desktop/issues/2485 to track this improvement.

**Comment by shiftkey** [comment]
@joshaber yeah, I figured there was some back story. :shipit:

**Comment by joshaber** [comment]
>It works mostly, but still not always:

Huh, that is interesting. I thought we were using the same heuristics but apparently not 🤔 I opened https://github.com/desktop/desktop/issues/2700 to track this.

>Also, in my opinion, the color saturation is way better in GitHub Desktop.

Yeah, we agree, we're tracking that in https://github.com/desktop/desktop/issues/2586.

**Pull request merged**